src/app/collective-rewards/rewards/components/ClaimRewardModal/ClaimRewardsModalView.tsx

```diff
@@ -0,0 +1,167 @@
+import { TransactionInProgressButton } from '@/app/user/Stake/components/TransactionInProgressButton'
+import { Button } from '@/components/ButtonNew'
+import { LoadingSpinner } from '@/components/LoadingSpinner'
+import { Modal } from '@/components/Modal/Modal'
+import { TokenImage, TokenSymbol } from '@/components/TokenImage'
+import { Tooltip } from '@/components/Tooltip'
+import { Typography } from '@/components/TypographyNew/Typography'
+import { RBTC, RIF } from '@/lib/constants'
+import { cn, formatCurrency } from '@/lib/utils'
+import * as RadioGroup from '@radix-ui/react-radio-group'
+import { FC, ReactNode, memo } from 'react'
+import { ClaimRewardType } from './types'
+
+interface ClaimRewardsModalViewProps {
+  onClose: () => void
+  selectedRewardType: ClaimRewardType
+  onRewardTypeChange: (value: ClaimRewardType) => void
+  rifAmount: bigint
+  rbtcAmount: bigint
+  rifFiatAmount: number
+  rbtcFiatAmount: number
+  totalFiatAmount: number
+  onClaim: () => void
+  isClaimable: boolean
+  isTxPending?: boolean
+  isLoading?: boolean
+  className?: string
+}
+
+interface ClaimRewardRadioOptionProps {
+  value: ClaimRewardType
+  label: ReactNode
+  subLabel: ReactNode
+  isLoading?: boolean
+}
+
+const ClaimRewardRadioOption: FC<ClaimRewardRadioOptionProps> = memo(
+  ({ value, label, subLabel, isLoading }) => (
+    <RadioGroup.Item
+      value={value}
+      className="flex-1 rounded border-1 border-[v3-text-100] px-3 py-4 flex flex-col items-start gap-2 data-[state=checked]:border-t-4 data-[state=checked]:pt-[13px] focus:outline-none cursor-pointer"
+    >
+      {isLoading ? (
+        <div className="flex justify-center items-center w-full py-2">
+          <LoadingSpinner size="small" />
+        </div>
+      ) : (
+        <div className="flex flex-row items-start gap-3 w-full">
+          <span className="w-5 h-5 aspect-square rounded-full border-2 border-white flex items-center justify-center flex-shrink-0">
+            <RadioGroup.Indicator className="w-full h-full rounded-full border-4 border-white" />
+          </span>
+          <div className="flex flex-col items-start gap-2 justify-start text-left w-full">
+            <Typography variant="h3">{label}</Typography>
+            <Typography variant="body" className="text-v3-bg-accent-0">
+              {subLabel}
+            </Typography>
+          </div>
+        </div>
+      )}
+    </RadioGroup.Item>
+  ),
+)
+
+ClaimRewardRadioOption.displayName = 'ClaimRewardRadioOption'
+
+export const ClaimRewardsModalView: FC<ClaimRewardsModalViewProps> = ({
+  onClose,
+  selectedRewardType,
+  onRewardTypeChange,
+  rifAmount,
+  rbtcAmount,
+  rifFiatAmount,
+  rbtcFiatAmount,
+  totalFiatAmount,
+  onClaim,
+  isClaimable,
+  isTxPending = false,
+  isLoading = false,
+  className,
+}) => {
+  const radioOptions: Array<{
+    value: ClaimRewardType
+    label: ReactNode
+    subLabel: ReactNode
+  }> = [
```

Comment:
I would also have this dynamic, instead of having it hardcoded Ill pass down an array